### PR TITLE
Remove OpenModelicaScriptingAPI.tmp.mo after use

### PR DESCRIPTION
**Pull request by phannebohm**
After building (with cmake at least) the .tmp.mo file pollutes the working tree of git. Removing it seems to be fine.

### CODE REVIEW

**Comment by mahge** [comment]
I have noticed this as well and it causes more issues than just polluting the git working tree. Sometimes it puts the compilation in a position where it will repeatedly invokes `OpenModelicaScriptingAPI.mos` wherever you try to compile omc even though the file is not actually modified. Sometimes the file is removed as it is supposed to be. Sometimes it is not. I am not sure what the root issue is but it is annoying because you have to manually delete this file, `OpenModelicaScriptingAPI.mo`, `OpenModelicaScriptingAPIQt.cpp`, and `OpenModelicaScriptingAPIQt.h` to avoid this repeated invocation of  `OpenModelicaScriptingAPI.mos`.

In any case, the scripting function `OpenModelica.Scripting.compareFilesAndMove` is supposed to compare two files and then

- Move first to the second if their contents are different OR  
- Remove the first file if their contents are the same. 

as you can see here
https://github.com/OpenModelica/OpenModelica/blob/2cbc275421af56c4c6bf8451e92f0e4bbe795c0f/OMCompiler/Compiler/Script/CevalScript.mo#L1114-L1120 

So the line https://github.com/OpenModelica/OpenModelica/blob/2cbc275421af56c4c6bf8451e92f0e4bbe795c0f/OMCompiler/Compiler/Script/OpenModelicaScriptingAPI.mos#L8

should have removed `OpenModelicaScriptingAPI.tmp.mo` in either case. There is something going wrong with `OpenModelica.Scripting.compareFilesAndMove`

**Review by mahge — APPROVED** [review]

File: OMCompiler/Compiler/Script/OpenModelicaScriptingAPI.mos [hunk 1]
```diff
@@ -6,6 +6,7 @@ if bool then
   bool := loadFile("OpenModelicaScriptingAPI.tmp.mo");
   if bool then
     bool := OpenModelica.Scripting.compareFilesAndMove("OpenModelicaScriptingAPI.tmp.mo", "OpenModelicaScriptingAPI.mo");
+    OpenModelica.Scripting.remove("OpenModelicaScriptingAPI.tmp.mo");
```
Comment (mahge) [inline OMCompiler/Compiler/Script/OpenModelicaScriptingAPI.mos:9]: I just checked the definition for `remove` and it looks like it is intended for removing directories. I am not sure what happens when the argument is a file.

https://github.com/OpenModelica/OpenModelica/blob/2cbc275421af56c4c6bf8451e92f0e4bbe795c0f/OMCompiler/Compiler/Script/CevalScript.mo#L1045-L1046

Comment (mahge) [reply OMCompiler/Compiler/Script/OpenModelicaScriptingAPI.mos:9]:
```suggestion
    OpenModelica.Scripting.removeFile("OpenModelicaScriptingAPI.tmp.mo");
```

**Comment by phannebohm** [comment]
I don't understand why Jenkins keeps failing, the first commit worked... @mahge any idea?

**Comment by mahge** [comment]
I have no idea. Does it work for you locally?

**Comment by mahge** [comment]
My bad. I suggested `OpenModelica.Scripting.removeFile` but there is no such function. I think it should be `OpenModelica.Scripting.deleteFile` instead.

**Comment by mahge** [comment]
I tested this with a fresh OpenModelica clone (after trying to duplicate it with my existing clone). It compiles without issues. I am not sure what is happening on Jenkins. Now that I think about it, it also compiled just fine on Jenkins with the current changes on the branch (force pushed) originally. 

I think there is something wrong with the CI. I will restart the build and we will see what happens,

**Pull request closed without merging**